docs/adr/0003-package-manager-guide.md

```diff
@@ -0,0 +1,146 @@
+    dependencies.
+  * Generates or updates configuration files so that the package manager builds with Hermeto's pre-
+    fetched dependencies.
+* Maintainers are satisfied with the design document details.
+* Demonstration of a proof of concept hermetic build for the package manager. The PoC does not need
+  to use a "real" project - a "hello world" project with minimal dependencies is sufficient.
+
+#### Incubating Phase
+
+During the incubating phase, the package manager should still be disabled by default, but does not
+need to emit any "experimental" warnings. The Hermeto community should seek feedback from the
+package manager's community regarding the process and demonstrate hermetic builds with increasing
+levels of project complexity.
+
+Incubating package managers should use this phase to draft end user documentation (under the
+`docs/` directory). Documentation should instruct end users how to generate the dependency list and
+execute a hermetic build with Hermeto. These procedures can reference external tools that are proof
+of concept or otherwise not fully supported by the package manager's community. If such a tool is
+utilized, the documentation should indicate these procedures are subject to change at any time.
+
+Incubating package managers are allowed to make breaking changes. If a breaking change is
+introduced, it should be documented in a release note.
+
+Incubating package managers are promoted to "Graduated" when the Hermeto community agrees the
+following criteria are satisfied:
+
+1. The process for generating the dependency list is supported by the underlying package manager's
+   community - either directly or through a well-maintained external party tool.
+2. A developer with reasonable package manager experience can use Hermeto to pre-fetch dependencies
+   by reading the project documentation.
+2. A sufficient number of demonstration/pilot projects have been able to use Hermeto to execute a
+   hermetic build with the package manager.
+
+#### Graduated Phase
+
+Graduated package managers are moved from the `_dev_package_managers` set to the full set of
+supported package managers. At this point the package manager is enabled by default and suitable
+for general use.
+
+Graduated package managers can only be removed with a major version update to Hermeto (ex: v1 to
+v2). User-facing details such as configuration values/flags can be deprecated and rendered non-
+functional with a minor version update, but cannot be removed without a major version update. Any
+deprecation must include a release note announcing the change.
+
+## Consequences
+
+* Contribution guidelines will need to be updated to codify the process/lifecycle in this ADR,
+  including:
+  * Iterative design documentation alongside code changes.
+  * Documentation requirements for promoting/graduating package managers.
+* Our current support for RPM/DNF should be documented (technical designs and end-user procedures).
```

Comment:
Do we still need to do anything about this with https://github.com/hermetoproject/hermeto/pull/1024 having been merged?